bpf/lib/trace_helpers.h

```diff
@@ -0,0 +1,74 @@
+/* SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause) */
+/* Copyright Authors of Cilium */
+
+#ifndef TRACE_ID_UTIL_H
+#define TRACE_ID_UTIL_H
+
+#include <bpf/ctx/ctx.h>
+#include <bpf/api.h>
+#include "common.h"
+#include "ip_options.h"
+
+#ifdef IP_TRACING_OPTION_TYPE
+/* Define the ip trace ID map with __u64 trace_id */
+struct {
+	__uint(type, BPF_MAP_TYPE_PERCPU_ARRAY);
+	__uint(max_entries, 1);
+	__type(key, __u32); /* only one key */
+	__type(value, __u64); /* trace_id type */
+	__uint(pinning, LIBBPF_PIN_BY_NAME);
+} trace_id_map __section_maps_btf;
+
+/* bpf_trace_id_set sets the trace_id in the map. */
+static __always_inline __u64 bpf_trace_id_set(__u64 trace_id)
+{
+	__u32 __z = 0;
```

Comment:
nit: I don't think it's part of our C convention to prefix `__`  onto internal var1 names.

Reply:
Thanks, let me push a change for that.